### PR TITLE
Feature: Consume Discord Role Removal API in Real Dev Squad Backend

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
### Description:
This pull request aims to enhance the Real Dev Squad backend by integrating and consuming the previously developed Discord Role Removal API [Real-Dev-Squad/discord-slash-commands/pull/82] . The integration allows the backend to utilize the API's functionality to remove roles from users in Discord seamlessly.

### Key Changes:
1. **API Integration**: The Real Dev Squad backend has been updated to integrate with the Discord Role Removal API. 

2. **API Request**: The backend can now send a DELETE request to the Discord Role Removal API when triggered by the corresponding action or event. The request includes the user and role identifiers required to perform the role removal operation in Discord.

3. **Error Handling**: Error handling with try and catch block will be maintained in the function consuming this service.

4. **Logging and Monitoring**: Logging and monitoring have been enhanced within the backend to capture relevant information and track the integration with the Discord Role Removal API. This allows for easier debugging, performance monitoring, and error tracking.

Testing:
Comprehensive testing has been performed to ensure the correctness and reliability of the integration. The tests cover various use cases, including successful role removal, error scenarios, and edge cases. Mocking techniques have been utilized to simulate API responses and validate the backend's behavior accurately.

Test Coverage: 
image

Please review this pull request and provide your feedback, suggestions, and any concerns you may have regarding the integration of the Discord Role Removal API into the Real Dev Squad backend.

### CODE REVIEW

File: services/discordService.js [hunk 1]
```diff
@@ -51,8 +51,25 @@ const addRoleToUser = async (userid, roleid) => {
   return response;
 };
 
+const removeRoleFromUser = async (roleId, discordId) => {
```
Comment (shubham-y) [inline services/discordService.js:54]: Can you please point out the place where we are using this function?

File: services/discordService.js [hunk 2]
```diff
@@ -51,8 +51,25 @@ const addRoleToUser = async (userid, roleid) => {
...
+  try {
+    const authToken = await generateAuthTokenForCloudflare();
+    const data = await fetch(`${DISCORD_BASE_URL}/roles`, {
+      method: "DELETE",
+      headers: { "Content-Type": "application/json", Authorization: `Bearer ${authToken}` },
+      body: JSON.stringify({ userid: discordId, roleid: roleId }),
```
Comment (shubham-y) [inline services/discordService.js:60]: I suggest we should update the API to expect these IDs as `userId` and `roleId`

File: services/discordService.js [hunk 3]
```diff
@@ -51,8 +51,25 @@ const addRoleToUser = async (userid, roleid) => {
...
+    });
+    const response = await data.json();
+    return response;
+  } catch (err) {
+    logger.error("Error in making fetch call to remove the role", err);
```
Comment (shubham-y) [inline services/discordService.js:65]: I think having a general message is better as error can occur due to any other operation as well and not only due to fetch

File: test/unit/services/discordService.test.js [hunk 4]
```diff
@@ -73,4 +78,39 @@ describe("Discord services", function () {
...
+      );
+      const response = await removeRoleFromUser("112233445566778899", "987654321123456789");
+      expect(response).to.deep.equal({
+        message: "Role Removed Successfully",
+        userAffected: { userid: "987654321123456789", roleid: "112233445566778899" },
+      });
+      expect(fetchStub.calledOnce).to.be.equal(true);
+    });
+
+    it("makis a failing fetch call to discord", async function () {
```
Comment (shubham-y) [inline test/unit/services/discordService.test.js:108]:
```suggestion
    it("makes a failing fetch call to discord", async function () {
```

File: services/discordService.js (hunk 2 above)
Comment (RitikJaiswal75) [reply services/discordService.js:60]: We cant do that because the discord uses userid and roleid so that requires a long chain of change

File: services/discordService.js (hunk 1 above)
Comment (RitikJaiswal75) [reply services/discordService.js:54]: We are not using it now will be using it in subsequent PRs as this pr just creates a service we will be using it at multiple places in near future

**Review by shubham-y — APPROVED** [review]

**Review by heyrandhir — APPROVED** [review]
lgtm
